api/middleware/auth0Middleware.js

```diff
@@ -0,0 +1,43 @@
+const { expressjwt: jwt } = require('express-jwt');
+const jwksRsa = require('jwks-rsa');
+const createError = require('http-errors');
+const config = require('../../config/auth0');
+
+// writing the config for expressJwt method to be invoked with product environment variables
+const verifyJwt = jwt({
+  secret: jwksRsa.expressJwtSecret({
+    cache: true,
+    rateLimit: true,
+    jwksRequestsPerMinute: 5,
+    jwksUri: config.jwksUri,
+  }),
+  audience: config.audience,
+  issuer: config.issuer,
+  algorithms: ['RS256'],
+  requestProperty: 'auth0User',
+}).unless({ path: ['/'] });
+//************ */
+//in the above line add all public endpoints (routes with no need for auth) as a string inside the path array [] <=
+// **** IMPORTANT note: if your public endpoint has descendent end points (ex.: ./yourRouterEndPoint/:user_id) you need ot use regex as in the following example:
+// ({ path: ['/', /^\/<your router end point>\/.*/] }); => example: ({ path: ['/', /^\/application\/.*/, /^\/roles\/.*/] });
```

Comment:
I love that you included an explanation with examples of how to add descendent endpoints selectively. Great use of inline comments!